cypress/integration/mocked-okta/registerController.1.cy.ts

```diff
@@ -0,0 +1,91 @@
+import userStatuses from '../../support/okta/userStatuses';
+import userResponse from '../../fixtures/okta-responses/success/user.json';
+import userExistsError from '../../fixtures/okta-responses/error/user-exists.json';
+import successTokenResponse from '../../fixtures/okta-responses/success/token.json';
+import resetPasswordResponse from '../../fixtures/okta-responses/success/reset-password.json';
+
+beforeEach(() => {
+  cy.mockPurge();
+});
+const verifyInRegularEmailSentPage = () => {
+  cy.contains('Check your email inbox');
+  cy.contains('Resend email');
+};
+
+userStatuses.forEach((status) => {
+  context(`Given I am a ${status || 'nonexistent'} user`, () => {
+    context('When I submit the form on /register', () => {
+      beforeEach(() => {
+        cy.visit(`/register?useOkta=true`);
+        cy.get('input[name="email"]').type('example@example.com');
+      });
+      switch (status) {
+        case false:
+          specify(
+            "Then I should be shown the 'Check your email inbox' page",
+            () => {
+              // Set the correct user status on the response
+              const response = { ...userResponse.response, status };
+              cy.mockNext(userResponse.code, response);
+              cy.get('button[type=submit]').click();
+              verifyInRegularEmailSentPage();
+            },
+          );
+          break;
+        case 'ACTIVE':
```

Comment:
I don't know what we would do here to test register with social users ? @coldlink  do we need to mock pushing the apple, facebook and google button instead of submitting the form ?

Reply:
I think the social users case we're talking about here is if a social only user already exists in the okta user store in the ACTIVE state but without a password.

So we're talking about an existing user attempting to register with an email that already exists. In this case the ACTIVE case covers both the social and non-social existing accoubnt

Reply:
I could do what I did with other tests and add a second version of the test but use a social user json response. The only difference is the contents of the social part of the json that I don't think anything checks right now.

Reply:
I ended up adding the social user case like I did everywhere else to complete the way the rest of the tests are written. I guess it validates that the code on our side wouldn't do anything different if the user response has different credentials